WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift

```diff
@@ -0,0 +1,138 @@
+import Gridicons
+import UIKit
+
+/// A full-width banner view to be shown at the top of a tab below the navigation bar.
+/// Consists of an icon, text label, action button and dismiss button.
+///
+final class TopBannerView: UIView {
+    private lazy var iconImageView: UIImageView = {
+        let imageView = UIImageView(image: nil)
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var infoLabel: UILabel = {
+        let label = UILabel(frame: .zero)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    private lazy var dismissButton: UIButton = {
+        let button = UIButton(frame: .zero)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    private lazy var actionButton: UIButton = {
+        let button = UIButton(frame: .zero)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    private let onDismiss: () -> Void
+    private let onAction: () -> Void
+
+    init(viewModel: TopBannerViewModel) {
+        onDismiss = viewModel.dismissHandler
+        onAction = viewModel.actionHandler
+        super.init(frame: .zero)
+        configureSubviews()
+        configureSubviews(viewModel: viewModel)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension TopBannerView {
+    func configureSubviews() {
+        backgroundColor = .white
+
+        let textStackView = UIStackView(arrangedSubviews: [titleLabel, infoLabel])
```

Comment:
I would consider splitting configuring each view into its own method, that would, in my opinion make this method more readable, as it seems a bit complicated, at least to me, to figure out what is included into what. So it would read something like:

```
func configureSubviews() {
	configureBackground()
	let contentStackView = configureContentStackView()
	let containerStackView = configureContainerStackView(content: contentStackView)
	let outerStackView = configureOuterStackView(container: containerStackView)
}
``

Reply:
until we have SwiftUI... will update!